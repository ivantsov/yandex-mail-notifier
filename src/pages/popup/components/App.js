import PropTypes from 'prop-types';
import React, {Component} from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';
import UnavailableMessage from './UnavailableMessage/UnavailableMessage';
import Donation from './Donation/Donation';
import Header from './Header/Header';
import List from './List/List';

class App extends Component {
    static propTypes = {
        user: PropTypes.shape({
            authorized: PropTypes.bool.isRequired,
            email: PropTypes.string.isRequired,
        }).isRequired,
        messages: PropTypes.shape({
            unreadCount: PropTypes.number.isRequired,
            items: PropTypes.arrayOf(PropTypes.object).isRequired,
            loading: PropTypes.bool.isRequired,
            error: PropTypes.bool.isRequired,
        }).isRequired,
        settings: PropTypes.shape({
            preferredDomain: PropTypes.string.isRequired,
        }).isRequired,
        loadMessages: PropTypes.func.isRequired,
        updateMessage: PropTypes.func.isRequired,
        openLink: PropTypes.func.isRequired,
        openSettings: PropTypes.func.isRequired,
        reloadApp: PropTypes.func.isRequired,
        openDonationLink: PropTypes.func.isRequired,
    };

    componentDidMount() {
        const {
            user,
            loadMessages,
        } = this.props;

        if (user.authorized) {
            loadMessages();
        }
    }

    render() {
        const {
            user,
            messages: {
                unreadCount,
                items,
                loading,
                error,
            },
            settings: {
                preferredDomain,
            },
            loadMessages,
            updateMessage,
            openLink,
            openSettings,
            reloadApp,
            openDonationLink,
        } = this.props;

        if (!user.authorized) {
            return (
                <UnavailableMessage
                    domain={preferredDomain}
                    reloadApp={reloadApp}
                />
            );
        }

        return (
            <div>
                <Header
                    user={user.email}
                    unreadMessagesCount={unreadCount}
                    disabled={loading}
                    reloadMessages={loadMessages}
                    openLink={openLink}
                    openSettings={openSettings}
                />
                <List
                    loading={loading}
                    error={error}
                    unreadMessagesCount={unreadCount}
                    items={items}
                    onActionClick={updateMessage}
                    openMessage={openLink}
                />
                <Donation onClick={openDonationLink}/>
            </div>
        );
    }
}

export default connect(state => state, actions)(App);
export {App as AppComponent}; // for test purpose
