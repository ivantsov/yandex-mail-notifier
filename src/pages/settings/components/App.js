import PropTypes from 'prop-types';
import React from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';
import config from './config';
import Field from './Field/Field';

const App = ({settings, updateSettings}) => {
  const fields = config.map((item, index) => (
    <Field
      key={index}
      {...item}
      value={settings[item.name]}
      onChange={updateSettings}
    />
  ));

  return (
    <table>
      <tbody>{fields}</tbody>
    </table>
  );
};

App.propTypes = {
  settings: PropTypes.object.isRequired,
  updateSettings: PropTypes.func.isRequired,
};

export default connect(state => state, actions)(App);
export {App as AppComponent}; // for test purpose
