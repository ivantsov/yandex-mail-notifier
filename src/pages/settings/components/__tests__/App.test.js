import React from 'react';
import renderer from 'react-test-renderer';
import {AppComponent as App} from '../App';

jest.mock('shared/utils/i18n');
jest.mock('../Field/Field', () => (props) => <div {...props}>Field</div>);

it('settings/App', () => {
    const component = renderer.create(
        <App
            settings={{
                newMessageNotification: 0,
                unreadMessagesNotification: 0,
                unreadMessagesSound: false,
                notAuthNotification: 0,
                preferredDomain: 'ru',
            }}
            updateSettings={jest.fn()}
        />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
