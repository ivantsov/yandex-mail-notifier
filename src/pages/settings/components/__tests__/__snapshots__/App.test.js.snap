// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings/App 1`] = `
<table>
  <tbody>
    <div
      description="settings.newMessageNotification.description"
      label="settings.newMessageNotification.label"
      name="newMessageNotification"
      onChange={[Function]}
      optionValues={
        Array [
          0,
          1,
          2,
        ]
      }
      options={
        Array [
          Object {
            "label": "settings.newMessageNotification.options.0",
            "value": 0,
          },
          Object {
            "label": "settings.newMessageNotification.options.1",
            "value": 1,
          },
          Object {
            "label": "settings.newMessageNotification.options.2",
            "value": 2,
          },
        ]
      }
      type="select"
      value={0}
    >
      Field
    </div>
    <div
      description="settings.unreadMessagesNotification.description"
      label="settings.unreadMessagesNotification.label"
      name="unreadMessagesNotification"
      onChange={[Function]}
      optionValues={
        Array [
          0,
          5,
          15,
          30,
        ]
      }
      options={
        Array [
          Object {
            "label": "settings.unreadMessagesNotification.options.0",
            "value": 0,
          },
          Object {
            "label": "settings.unreadMessagesNotification.options.1",
            "value": 5,
          },
          Object {
            "label": "settings.unreadMessagesNotification.options.2",
            "value": 15,
          },
          Object {
            "label": "settings.unreadMessagesNotification.options.3",
            "value": 30,
          },
        ]
      }
      type="select"
      value={0}
    >
      Field
    </div>
    <div
      description="settings.unreadMessagesSound.description"
      label="settings.unreadMessagesSound.label"
      name="unreadMessagesSound"
      onChange={[Function]}
      options={undefined}
      type="checkbox"
      value={false}
    >
      Field
    </div>
    <div
      description="settings.notAuthNotification.description"
      label="settings.notAuthNotification.label"
      name="notAuthNotification"
      onChange={[Function]}
      optionValues={
        Array [
          0,
          1,
          2,
        ]
      }
      options={
        Array [
          Object {
            "label": "settings.notAuthNotification.options.0",
            "value": 0,
          },
          Object {
            "label": "settings.notAuthNotification.options.1",
            "value": 1,
          },
          Object {
            "label": "settings.notAuthNotification.options.2",
            "value": 2,
          },
        ]
      }
      type="select"
      value={0}
    >
      Field
    </div>
    <div
      description="settings.preferredDomain.description"
      label="settings.preferredDomain.label"
      name="preferredDomain"
      onChange={[Function]}
      optionValues={
        Array [
          "ru",
          "ua",
          "by",
          "kz",
          "com",
          "com.tr",
        ]
      }
      options={
        Array [
          Object {
            "label": "settings.preferredDomain.options.0",
            "value": "ru",
          },
          Object {
            "label": "settings.preferredDomain.options.1",
            "value": "ua",
          },
          Object {
            "label": "settings.preferredDomain.options.2",
            "value": "by",
          },
          Object {
            "label": "settings.preferredDomain.options.3",
            "value": "kz",
          },
          Object {
            "label": "settings.preferredDomain.options.4",
            "value": "com",
          },
          Object {
            "label": "settings.preferredDomain.options.5",
            "value": "com.tr",
          },
        ]
      }
      type="select"
      value="ru"
    >
      Field
    </div>
    <div
      description="settings.setShortcuts.description"
      label="settings.setShortcuts.label"
      name="setShortcuts"
      onChange={[Function]}
      options={undefined}
      text="settings.setShortcuts.linkText"
      type="link"
      url="chrome://extensions/configureCommands"
      value={undefined}
    >
      Field
    </div>
  </tbody>
</table>
`;
